avocado/core/safeloader.py

```diff
@@ -497,8 +497,8 @@ def _examine_class(path, class_name, match, target_module, target_class,
 
             modules_paths = [parent_path,
                              os.path.dirname(module.path)] + sys.path
-            _, found_ppath, _ = imp.find_module(parent_module,
-                                                modules_paths)
+            found_ppath = PathFinder.find_spec(parent_module,
+                                               modules_paths).origin
```

Comment:
IIUC, according to the documentation, this method can return None, and you trying to access the `.origin` property of a `None` value. This will raise exceptions.

You should properly check if the return is `None` or not, before using.

Reply:
Hi @beraldoleal, thanks for pointing that out :man_facepalming:, you are absolutely right. I will add an if statement here to call `_examine_class` only if `find_spec` does not return `None`.